pylint/checkers/copyright-checker.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# © Copyright IBM Corporation 2020.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+
+from pylint.interfaces import IRawChecker, HIGH
+from pylint.checkers import BaseChecker
+
+
+class CopyrightChecker(BaseChecker):
+    """
+    Check for copyright
+    """
+
+    __implements__ = IRawChecker
+
+    name = 'copyright-checker'
+    msgs = {'E9902': ('Include or fix copyright in file',
+                      'file-no-copyright',
+                      'Your file has no copyright or it needs to be fixed'),
+            }
+    options = ()
+
+    def process_module(self, node):
+        ibm_copyright_regex = r'^# © Copyright IBM Corporation 20\d\d(?:, 20\d\d)?\.$'
```

Comment:
Can you extend this code to also accept the format of `cloudant_v1.py,` and `test_cloudant_v1.py`?
There is `Corp.` used instead of `Corporation`, that is also acceptable form based on this: https://apps.na.collabserv.com/wikis/home?lang=en-us#Wording of copyright notices:~:text=%C2%A9%20Copyright%20IBM%20Corp.%201999%2C%202015.,-%C2%A9%20Copyright

Reply:
We are excluding generated files in other repos as well: https://github.com/IBM/cloudant-node-sdk/blob/master/tslint.json#L4 .

Also, I'm not sure what we'd gain from linting generated files. Can you please clarify?

Reply:
Just in this case you are creating a linter script, so you have bigger moving space.
Regarding the lawing rules it is always good if there is a header copyright in also the generated files.
It was just an idea.

Reply:
Hm, it actually sounds like a good idea. If we want to do that, we should do it in all languages. (And possibly for license headers, too.) Let's discuss this offline to avoid scope creep here.